app/components/recognition/form.js

```diff
@@ -332,11 +319,45 @@ export default class FormComponent extends Component {
   }
 
   @action
-  handleAwardedByChange(governingBody) {
-    this.currentRecognition.selectedItem = governingBody;
+  handleAwardedByChange(organizationName) {
+    this.currentRecognition.selectedItem = organizationName;
+    this.clearFormError('awardedBy');
+  }
+
+  @action
+  handleAwardedByChangeEvent(event) {
+    this.currentRecognition.recognitionModel.awardedBy = event.target.value;
     this.clearFormError('awardedBy');
   }
 
+  @action
+  async getAwardedBy() {
+    let awardedByValue;
+    if (this.currentRecognition.selectedItem == this.items[0]) {
+      // The user selected the option "College van burgemeester en schepenen"
+      // We want to link the recognition to the organization the user is logged in with
+      awardedByValue = this.currentSession.group.name;
```

Comment:
We already have the record in this scenario, so we don't need to fetch it again, right? I think we can early return here.